sources/packages/web/src/services/http/ApplicationApi.ts

```diff
@@ -55,4 +43,58 @@ export class ApplicationApi extends HttpBaseClient {
       throw error;
     }
   }
+
+  public async createApplicationDraft(
+    payload: SaveStudentApplicationDto,
+  ): Promise<number> {
+    try {
+      const response = await this.apiClient.post(
+        "application/draft",
+        payload,
+        this.addAuthHeader(),
+      );
+      return +response.data;
+    } catch (error) {
+      if (!error.response.data?.errorType) {
+        // If it is an not expected error,
+        // handle it the default way.
+        this.handleRequestError(error);
+      }
+
+      throw error;
+    }
+  }
+
+  public async saveApplicationDraft(
+    applicationId: number,
+    payload: SaveStudentApplicationDto,
+  ): Promise<number> {
+    try {
+      const response = await this.apiClient.patch(
+        `application/${applicationId}/draft`,
+        payload,
+        this.addAuthHeader(),
+      );
+      return +response.data;
+    } catch (error) {
+      this.handleRequestError(error);
+      throw error;
+    }
+  }
+
+  public async submitApplication(
+    applicationId: number,
+    payload: SaveStudentApplicationDto,
+  ): Promise<any> {
```

Comment:
I think here promise should be `void` instead of `any`. can you have look?

Reply:
true again, good catch 😉

Reply:
Changed!